### PR TITLE
Fixed issue #18343: Problem with interpreting double slash in URL (IIS)

### PR DESCRIPTION
**Pull request by Shnoulle**
Dev: use PHP to construct the url
Dev: check if non empty before adding it

### CODE REVIEW

**Comment by Shnoulle** [comment]
I do the master after this one is reviewed

**Comment by Shnoulle** [comment]
Alternative solution move to jquery with data-url and a on('click but finalyy it's same.

File: application/views/admin/survey/prepareEditorScript_view.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ function find_popup_editor(fieldname)
         return null;
     }
 
-    function start_popup_editor(fieldname, fieldtext, sid, gid, qid, fieldtype, action)
+    function start_popup_editor(fieldname, url)
```
Comment (gabrieljenik) [inline application/views/admin/survey/prepareEditorScript_view.php:52]: param could be called editorURL ? (a bit more descriptive)

**Review by gabrieljenik — APPROVED** [review]
Code looks good. Nice fix.

**Comment by Shnoulle** [comment]
> Code looks good. Nice fix.

I really think there are a better way …

File: application/views/admin/survey/prepareEditorScript_view.php (hunk 1 above)
Comment (Shnoulle) [reply application/views/admin/survey/prepareEditorScript_view.php]: Yep, and add some phpdoc

**Comment by Shnoulle** [comment]
Next step : use a dialog box + data-url etc …

**Comment by c-schmitz** [comment]
Only for 3.x?

**Comment by Shnoulle** [comment]
Oups …

**Comment by Shnoulle** [comment]
https://github.com/LimeSurvey/LimeSurvey/pull/2639

**Comment by Shnoulle** [comment]
Reporter test the commit : fixed the popup editor but have another issue with Placeholder for example.
https://bugs.limesurvey.org/view.php?id=18343#c72044

OK to set it as tested ?

**Comment by Shnoulle** [comment]
Done for LimeReplacementFields

File: application/helpers/admin/htmleditor_helper.php [hunk 2]
```diff
@@ -231,7 +268,27 @@ function getInlineEditor($fieldtype, $fieldname, $fieldtext, $surveyID = null, $
             ,filebrowserImageUploadUrl:'{$sFakeBrowserURL}'
             ,filebrowserFlashUploadUrl:'{$sFakeBrowserURL}'";
         }
-
+        /* @var string[] parameters of the replacementfields url */
```
Comment (Shnoulle) [inline application/helpers/admin/htmleditor_helper.php:271]: Move to a function in helper ?

**Pull request merged**